src/main/kotlin/net/irisshaders/lilybot/extensions/util/Tags.kt

```diff
@@ -0,0 +1,193 @@
+									"the following command:\n`/create-tag <name> <title> <value>`\nYou will be " +
+									"prompted to enter a name for the tag, a title for the tag, and the value for the" +
+									" tag. This is what will appear in the embed of your tag. You can enter any " +
+									"character you like into all of these inputs.\n\nTo use the tag, run the " +
+									"following command\n`/tag <name>`\nYou will be prompted to enter the tag name, " +
+									"but will have an autocomplete window to aid you. The window will list all the " +
+									"tags that the guild has.\n\nTo delete a tag, if you have the Moderate Members " +
+									"permission, run the following command\n`/delete-tag <name>`\nYou will be " +
+									"prompted to enter the name of the tag, again aided by autocomplete, and then " +
+									"the tag will be completed.\n\nGuilds can have any number of tags they like. The " +
+									"limit on `tagValue` for tags is 1024 characters, which is the embed description " +
+									"limit enforced by discord"
+						color = DISCORD_BLURPLE
+						timestamp = Clock.System.now()
+					}
+				}
+			}
+		}
+
+		ephemeralSlashCommand(::CreateTagArgs) {
+			name = "create-tag"
+			description = "Create a tag for your guild!"
+
+			check { anyGuild() }
+			check { hasPermission(Permission.ModerateMembers) }
+
+			action {
+				val actionLogId = getConfigPrivateResponse("modActionLog") ?: return@action
+				val actionLog = guild!!.getChannel(actionLogId) as GuildMessageChannelBehavior
+
+				actionLog.createEmbed {
+					title = "Tag created!"
+					description = "${user.asUser().mention} created the tag ${arguments.tagName}"
+					field {
+						name = "Tag contents:"
+						value = """
+							```text
+							title: ${arguments.tagTitle}
+							
+							value: ${arguments.tagValue}
+							```
+						""".trimIndent()
+						inline = false
+					}
+					footer {
+						icon = user.asUser().avatar?.url
+						text = "Requested by ${user.asUser().tag}"
+					}
+					timestamp = Clock.System.now()
+				}
+
+				DatabaseHelper.setTag(guild!!.id, arguments.tagName, arguments.tagTitle, arguments.tagValue)
+
+				respond {
+					content = "Tag: `${arguments.tagName}` created"
+				}
+			}
+		}
+
+		ephemeralSlashCommand(::TagArgs) {
+			name = "delete-tag"
+			description = "Delete a tag from your guild"
+
+			check { anyGuild() }
+			check { hasPermission(Permission.ModerateMembers) }
+
+			action {
+				if (DatabaseHelper.getTag(guild!!.id, arguments.tagName, "name") == null) {
+					respond {
+						content = "Unable to find tag! Does this tag exist?"
+					}
+					return@action
+				}
+
+				val actionLogId = getConfigPrivateResponse("modActionLog") ?: return@action
+				val actionLog = guild!!.getChannel(actionLogId) as GuildMessageChannelBehavior
+
+				responseEmbedInChannel(
+					actionLog,
+					"Tag deleted!",
+					"The tag ${arguments.tagName} was deleted by ${user.asUser().mention}",
+					DISCORD_RED,
+					user.asUser()
```

Comment:
Probably don't need the user bit if we're using the footer. Should we log the contents of the tag or is that redundant as it'll be logged on creation/update?

Reply:
Redundant imo